### PR TITLE
S2696 - update tests

### PR DESCRIPTION
**Pull request by andrei-epure-sonarsource**
IMO static lambdas should work as they currently work.

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.CSharp/Rules/StaticFieldWrittenFromInstanceMember.cs [hunk 1]
```diff
@@ -31,34 +31,18 @@ namespace SonarAnalyzer.Rules.CSharp
...
-                {
-                    return false;
-                }
-            }
-
-            return true;
-        }
+        protected override bool IsValidCodeBlockContext(SyntaxNode node, ISymbol owningSymbol) =>
+            (owningSymbol != null && !owningSymbol.IsStatic)
+            && (node is MethodDeclarationSyntax || node is AccessorDeclarationSyntax);
```
Comment (andrei-epure-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Rules/StaticFieldWrittenFromInstanceMember.cs:45]: Could this be done before getting the symbol

Comment (andrei-epure-sonarsource) [reply analyzers/src/SonarAnalyzer.CSharp/Rules/StaticFieldWrittenFromInstanceMember.cs:45]: we get the symbol from the context, all good

**Comment by andrei-epure-sonarsource** [comment]
I tried to add coverage for the defensive coding but could not.

File: analyzers/src/SonarAnalyzer.CSharp/SonarAnalyzer.CSharp.csproj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <LangVersion>8.0</LangVersion>
+    <LangVersion>9.0</LangVersion>
```
Comment (andrei-epure-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/SonarAnalyzer.CSharp.csproj:4]: I will change the other projects lang version (and move it to Directory.Build.props) in a future PR

File: analyzers/src/SonarAnalyzer.CSharp/Rules/StaticFieldWrittenFromInstanceConstructor.cs [hunk 3]
```diff
@@ -32,31 +32,17 @@ namespace SonarAnalyzer.Rules.CSharp
...
-            }
-
-            return true;
-        }
-
-        protected override string GetDiagnosticMessageArgument(SyntaxNode node, ISymbol owningSymbol, IFieldSymbol field)
-        {
-            return field.Name;
-        }
+        protected override string GetDiagnosticMessageArgument(SyntaxNode node, ISymbol owningSymbol, IFieldSymbol field) => field.Name;
```
Comment (csaba-sagi-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Rules/StaticFieldWrittenFromInstanceConstructor.cs:46]: If I interpret our coding style correctly this should go to a new line.
"Expression body of method should be on the same line only for trivial cases: literal or simple identifier. Member access, indexer, invocation and other complex structures should be on the next line."

File: analyzers/src/SonarAnalyzer.CSharp/Rules/StaticFieldWrittenFromInstanceMember.cs [hunk 4]
```diff
@@ -31,34 +32,18 @@ namespace SonarAnalyzer.Rules.CSharp
...
-                if (declaration == null)
-                {
-                    return false;
-                }
-            }
-
-            return true;
-        }
+        protected override bool IsValidCodeBlockContext(SyntaxNode node, ISymbol owningSymbol) =>
+            (owningSymbol != null && !owningSymbol.IsStatic)
```
Comment (csaba-sagi-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Rules/StaticFieldWrittenFromInstanceMember.cs:45]:
```suggestion
           owningSymbol is { IsStatic: false }
```

**Review by csaba-sagi-sonarsource — APPROVED** [review]
LGTM! I left two minor suggestions.

File: analyzers/src/SonarAnalyzer.CSharp/Rules/StaticFieldWrittenFromInstanceConstructor.cs (hunk 3 above)
Comment (andrei-epure-sonarsource) [reply analyzers/src/SonarAnalyzer.CSharp/Rules/StaticFieldWrittenFromInstanceConstructor.cs:46]: I would consider this trivial enough...

**Pull request merged**